pkg/bpf/map.go

```diff
@@ -867,3 +867,29 @@ func (m *Map) resolveErrors() error {
 
 	return nil
 }
+
+// KeySize determines the BPF map key size for the specified size, by rounding
+// up the size according to the rules used by the kernel.
+//
+// Usage: bpf.KeySize(unsafe.Sizeof(foo{}))
+func KeySize(size uintptr) uint32 {
+	rem := size % 8
+	return uint32(size + (8 - rem))
```

Comment:
Without this logic, I was seeing the IPv6 CT map being removed/upgraded on startup (size 38 vs. 40), but with this logic it seems like the IPv4 CT map is being removed/upgraded (size 14 vs. 16). Need to dig into whether this is strictly necessary or not.

Reply:
Turns out the IPv4 CT tuple key is backed on the BPF side, but the IPv6 one isn't. @borkmann any preference on packed vs. not?

Reply:
I opted to pack the ipv6 tuple, since that seems to make the C definitions of the key match the Golang definitions.